### PR TITLE
Failed async should not block when the test is failed - fixes #58

### LINKED ISSUE #58: Failed async should not block when the test is failed
I reproduced this issue in a minimal test:
https://gist.github.com/jotak/3779cb4a8fc9a23cfdafd970cc386214

As you can see I have to surround my assertions in `try/finally` in order to get the async completed upon failure. Another workaround would be to set a timeout ( `async.await(5000)` ), but still I would expect the test to fail immediately on assertion failure.

I haven't checked the internals of `TestContext`, but I would expect all created `Async` objects to be completed automatically on failure.

### PR DESCRIPTION
**Pull request by vietj**
Can you have a look and check @jotak @bhowell2 ?

this should be a fix for your case.

### CODE REVIEW

**Comment by vietj** [comment]
note : I've simplified the implementation of the `Step` class so the `running` flag is not used anymore and instead an `Async` is created and completed by the thread executing the test. Consequently any reported failure makes a test fail fast.

File: src/test/java/io/vertx/ext/unit/TestSuiteNoEventLoopTest.java [hunk 1]
```diff
@@ -28,6 +28,7 @@ protected boolean checkTest(TestContext test) {
     return Vertx.currentContext() == null;
   }
 
+  /*
```
Comment (jotak) [inline src/test/java/io/vertx/ext/unit/TestSuiteNoEventLoopTest.java]: So this test doesn't make sense now? Or should it be adapted in a different way?

Comment (vietj) [reply src/test/java/io/vertx/ext/unit/TestSuiteNoEventLoopTest.java]: it does not make sense, I left it as commented to revisit it later

**Comment by jotak** [comment]
Ok, looks good to me

**Comment by bhowell2** [comment]
Looks good to me too

**Pull request merged**